aten/src/ATen/gen_backend_select_register.py

```diff
@@ -0,0 +1,48 @@
+from code_template import CodeTemplate
+
+FUNCTION_REGISTRATION = CodeTemplate("""\
+.op(torch::RegisterOperators::options()
+  .schema("${schema_string}")
+  .impl_unboxedOnlyKernel<decltype(${function_name}), &${function_name}>(DispatchKey::BackendSelect)
+  .aliasAnalysis(AliasAnalysisKind::FROM_SCHEMA))
+""")
+
+FUNCTION_DEFINITION = CodeTemplate("""\
+Tensor ${function_name}(${method_formals}) {
+  DispatchKey key = options.computeDispatchKey();
+  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::${name}", "${overload_name}");
+  return op.callUnboxedWithDispatchKey<${formals_types}>(key, ${type_method_actuals});
+}
+""")
+
+def register_backend_select_methods(declarations, template_path, file_manager):
+    backend_select_method_definitions = []
+    backend_select_function_registrations = []
+
+    for decl in declarations:
+        for option in decl["options"]:
+            if '_like' in option['name'] or 'new_' in option['name'] or 'to' in option['name']:
+                continue
+
+            name = option['name']
+            if 'overload_name' in option and option['overload_name'] != '':
+                name = "{0}_{1}".format(name, option['overload_name'])
+
+            if "arguments" in option and any("dynamic_type" in a and
+                                             a["dynamic_type"] == "TensorOptions" for a in option["arguments"]):
```

Comment:
This kind of pattern matching logic - where we're empirically identifying a subset of ops by secondary characteristics, rather than via a) some explicit declaration or b) a higher-level concept or property of the system - is another thing I think we should try to dial out of the CG pipeline as we go, to make it more robust and discoverable. 

@gchanan 's suggestion here was to trigger the codegen by a `backend_select` attribute instead. I can't think of a downside to this - it makes the new capability an explicit futureproofed thing that can be documented, composed with other things, etc.

Reply:
Added 'backend_select' key to native_function.yaml